docs/examples/101/api-management-modular/groups.bicep

```diff
@@ -0,0 +1,32 @@
+// ****************************************
+// Azure Bicep Module:
+// Create multiple groups from sample JSON records
+// ****************************************
+@minLength(1)
+param apimInstanceName string
+var groupsSet = [
+  {
+    'groupName': 'APIMGroup1'
+    'groupDisplayName': 'APIM Group 1'
+    'groupDescription': 'Description for this group'
+  }
+  {
+    'groupName': 'APIMGroup2'
+    'groupDisplayName': 'APIM Group 2'
+    'groupDescription': 'Description for this group'
+  }
+]
+
+//APIM Groups
+resource apimGroup 'Microsoft.ApiManagement/service/groups@2020-06-01-preview' = [for grp in groupsSet: {
+  name: '${apimInstanceName}/${grp.groupName}'

```

Comment:
While a bit tedious, it would be better to establish an `existing` refernece to the APIM resource you created in `main.bicep`, then reference that resource as the parent:

```bicep
resource apim 'Microsoft.ApiManagement/service@2020-06-01-preview' existing = {
  name: apimInstanceName

  resource apimGroup 'groups' = [ ... { 
    ... }]
}
```

Reply:
Clever idea. Done and rectified